resources/compilers/Linux/xml2st/PLCGenerator.py

```
@@ -0,0 +1,1782 @@
+                        break
+            while current_type is not None and len(parts) > 0:
+                blocktype = self.ParentGenerator.Controler.GetBlockType(current_type)
+                if blocktype is not None:
+                    name = parts.pop(0)
+                    current_type = None
+                    for var_name, var_type, _var_modifier in blocktype["inputs"] + blocktype["outputs"]:
+                        if var_name == name:
+                            current_type = var_type
+                            break
+                else:
+                    tagname = ComputeDataTypeName(current_type)
```

Ruff (0.8.2)

`ComputeDataTypeName` may be undefined, or defined from star imports

(F405)